File: kickstart/2022/Round B/infinity_area.c
#include <stdio.h>

double get_area(int r, int a, int b) {
    int current = r;
    const double pi = 3.14159265358979;
    double area = pi * current * current;

    while (current != 0) {
        current = current * a;
        area += (pi * current * current);
        
        current = current / b;
        area += (pi * current * current);
    }

    return area;
}

int main() {
    int testcases;
    scanf("%d", &testcases);
    int r, a, b;

    for (int t = 1; t <= testcases; ++t) {
            scanf("%d %d %d", &r, &a, &b);
            printf("Case #%d: %lf\n", t, get_area(r, a, b));
    }
    return 0;
}

File: kickstart/2022/Round B/infinity_area.py
from math import pi

def solve(r, a, b):
    current = r

    area = pi * current * current

    while current != 0:
        current = current * a
        area += (pi * current * current)

        current = current // b
        area += (pi * current * current)

    return area
    

if __name__ == '__main__':
    t = int(input())
    for case in range(1, t+1):
        r, a, b = map(int, input().split())
        result = solve(r, a, b)
        print('Case #{}: {}'.format(case, result))